src/NServiceBus.Core/Hosting/Helpers/AssemblyScanningComponent.cs

```diff
@@ -35,16 +35,19 @@ public static AssemblyScanningComponent Initialize(Configuration configuration,
             assemblyScanner.TypesToSkip = assemblyScannerSettings.ExcludedTypes;
             assemblyScanner.ScanNestedDirectories = assemblyScannerSettings.ScanAssembliesInNestedDirectories;
             assemblyScanner.ThrowExceptions = assemblyScannerSettings.ThrowExceptions;
+            assemblyScanner.ScanFileSystemAssemblies = assemblyScannerSettings.ScanFileSystemAssemblies;
             assemblyScanner.ScanAppDomainAssemblies = assemblyScannerSettings.ScanAppDomainAssemblies;
             assemblyScanner.AdditionalAssemblyScanningPath = assemblyScannerSettings.AdditionalAssemblyScanningPath;
 
             var scannableAssemblies = assemblyScanner.GetScannableAssemblies();
+            // Always scan the NServiceBus core because we rely on type discovery for internal features
+            assemblyScanner.AddTypesToResult(Assembly.GetExecutingAssembly(), scannableAssemblies);
```

Comment:
Would it be more descriptive and safer to always use something like

```
            assemblyScanner.AddTypesToResult(typeof(AssemblySCanningComponent).Assembly, scannableAssemblies);
```

?

Reply:
that would also work. Not sure I understand why it would be safer though?

Reply:
Safer is a bad way of me trying to express when I see GetExecutingAssembly I always have to think multiple times and what it actually does :D